test/integration/custom_targets.cpp

```diff
@@ -0,0 +1,85 @@
+#include "hsm/hsm.h"
+
+#include <boost/hana.hpp>
+#include <boost/hana/experimental/printable.hpp>
+#include <gtest/gtest.h>
+
+#include <future>
+#include <memory>
+
+namespace {
+
+using namespace ::testing;
+using namespace boost::hana;
+
+// States
+struct S1 {
+    std::string name = "S1";
+};
+struct S2 {
+    S2(std::string name)
+        : name(name)
+    {
+    }
+    std::string name;
+};
+
+// Events
+struct e1 {
+    std::string name = "e1";
+};
+struct e2 {
+};
+
+// Actions
+const auto stateFactory
+    = [](auto event, auto source) { return std::make_unique<S2>(event.name + source.name); };
+
+const auto log = [](auto event, auto source, auto target) {
+    std::cout << experimental::print(typeid_(source)) << ".name=" << source.name << " + "
+              << experimental::print(typeid_(event)) << " = "
+              << experimental::print(typeid_(target)) << ".name=" << target.name << std::endl;
+};
+
+struct MainState {
+    static constexpr auto make_transition_table()
+    {
+        // clang-format off
+        return hsm::transition_table(
+            * hsm::state<S1> {} + hsm::event<e1> {} / hsm::create(stateFactory) = hsm::state<S2> {}
+            , hsm::state<S2> {} + hsm::event<e2> {} / log = hsm::state<S1> {}
+            , hsm::state<S1> {} + hsm::event<e2> {} / hsm::reuse(log) = hsm::state<S2> {}
```

Comment:
Could you add some documentation about the difference between a plain lambda (`log`), a state factory `hsm::create(stateFactory)` and whatever `hsm::reuse(log)` is?

Reply:
- [ ] Add documentation